etc/contracts-test-data/contracts/mailbox/mailbox.sol

```diff
@@ -0,0 +1,103 @@
+    /// formats
+    /// @param data The calldata that is transmitted for the transaction call
+    /// @param signature An abstract set of bytes that are used for transaction authorization
+    /// @param factoryDeps The set of L2 bytecode hashes whose preimages were shown on L1
+    /// @param paymasterInput The arbitrary-length data that is used as a calldata to the paymaster pre-call
+    /// @param reservedDynamic The arbitrary-length field for usage in a future extension of transaction formats
+    struct L2CanonicalTransaction {
+        uint256 txType;
+        uint256 from;
+        uint256 to;
+        uint256 gasLimit;
+        uint256 gasPerPubdataByteLimit;
+        uint256 maxFeePerGas;
+        uint256 maxPriorityFeePerGas;
+        uint256 paymaster;
+        uint256 nonce;
+        uint256 value;
+        // In the future, we might want to add some
+        // new fields to the struct. The `txData` struct
+        // is to be passed to account and any changes to its structure
+        // would mean a breaking change to these accounts. To prevent this,
+        // we should keep some fields as "reserved"
+        // It is also recommended that their length is fixed, since
+        // it would allow easier proof integration (in case we will need
+        // some special circuit for preprocessing transactions)
+        uint256[4] reserved;
+        bytes data;
+        bytes signature;
+        uint256[] factoryDeps;
+        bytes paymasterInput;
+        // Reserved dynamic type for the future use-case. Using it should be avoided,
+        // But it is still here, just in case we want to enable some additional functionality
+        bytes reservedDynamic;
+    }
+
+    /// @notice New priority request event. Emitted when a request is placed into the priority queue
+    /// @param txId Serial number of the priority operation
+    /// @param txHash keccak256 hash of encoded transaction representation
+    /// @param expirationTimestamp Timestamp up to which priority request should be processed
+    /// @param transaction The whole transaction structure that is requested to be executed on L2
+    /// @param factoryDeps An array of bytecodes that were shown in the L1 public data.
+    /// Will be marked as known bytecodes in L2
+    event NewPriorityRequest(
+        uint256 txId,
+        bytes32 txHash,
+        uint64 expirationTimestamp,
+        L2CanonicalTransaction transaction,
+        bytes[] factoryDeps
+    );
+
+    function bridgehubRequestL2TransactionOnGateway(
+        L2CanonicalTransaction calldata _transaction,
+        bytes[] calldata _factoryDeps,
+        bytes32 _canonicalTxHash,
+        uint64 _expirationTimestamp
+    ) external {
+        _writePriorityOp(_transaction, _factoryDeps, _canonicalTxHash, _expirationTimestamp);
+    }
+
+    /// @notice Stores a transaction record in storage & send event about that
+    function _writePriorityOp(
```

Comment:
it does not write into the priority tree.

this operation would involving hashing the entire data + some storage writes

this makes the test less representative of the reality

Reply:
Would it be possible to reuse the existing mailbox?